### ISSUE TITLE
Absorption optimiser repeated log items

### ISSUE THREAD

**Issue by m0vva** [ESPresense/ESPresense-companion#387]
I have recently installed and configured ESPresense-companion and, apart from realising I need to add many more nodes, I am finding the following log item being generated 2-3 times per minute after ESPresense-companion optimiser has started running:

`[07:50:27 INF] Optimizer Absorption NelderMead    found worse  results, rms 179769313486232000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.000>179769313486232000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.000`

The values appear to be extreme.  Is this normal?  Can you recommend how to resolve it if is indeed irregular?

Thanks

**Comment by m0vva** [comment]
Thanks for your advice!

The floorpans are all set up here, so I will have to switch the optimisation back on and take a look at the Calibration UI, as suggested.  One concern I have is that some of my internal walls are thick (previously external) walls, which is likely to make an algorithmic calculation difficult.  I will add more ESP32s and monitor the Calibration UI.

I'm also mulling over if calibration could be done using machine learning but it's just a thought at the moment and unlikely to go anywhere quickly.

**Comment by DTTerastar** [comment]
yes, one idea is having an "absorption" map of cells, and then you could find the straight line path through the cells that made sense with all the node distance measurements.

**Comment by m0vva** [comment]
I'm getting no data reported in the calibration UI.

![Screenshot 2023-11-09 at 17 47 06](https://github.com/ESPresense/ESPresense-companion/assets/44472658/1c10d9c2-f4d5-4a29-a340-733c7c31f06e)

Is there something I can do to reset it and possibly remove those wild values in the log for Optimizer Rx Adj Rssi, Optimizer Absorption Avg and Optimizer NelderMead???

**Comment by DTTerastar** [comment]
Trying going into the console and looking at the json, it's probably got errors

**Comment by m0vva** [comment]
I just flashed one of of my nodes back to ESPresense v3.2.4 firmware and it appears in the Calibration UI, I'll re-flash all the other nodes as it appears that v3.3.1 breaks this functionality.

**Comment by DTTerastar** [comment]
I'm running the new and calibration is working so it must be a combination of factors.

**Comment by m0vva** [comment]
I think so too.  After re-adding my 7th node the Calibration UI has disappeared and those large results in the log have returned.  I'm not sure how to debug further - is there a log level I can set?

**Comment by m0vva** [comment]
@DTTerastar, I think I have figured out my mistake.  I had applied a filter to the node as per the ESPresense documentation to reduce the amount of MQTT messages flying around.  After removing the filter the optimisation starts to work again, although I think the MQTT broker is taking a hit.  I will check the code to see which messages you are subscribing to, I guess it's either node:* or iBeacon:*.  Hopefully I will be be able to apply a properly configured filter and get this working without error.
